DXMainClient/Online/FileHashCalculator.cs

```diff
@@ -94,11 +94,26 @@ public void CalculateHashes(List<GameMode> gameModes)
             #if !YR
             if (Directory.Exists(ProgramConstants.GamePath + "INI\\Map Code"))
             {
+                HashSet<string> extraINIFilenames = new HashSet<string>();
+
                 foreach (GameMode gameMode in gameModes)
                 {
                     fh.INIHashes = AddToStringIfFileExists(fh.INIHashes, "INI\\Map Code\\" + gameMode.Name + ".ini");
-                    Logger.Log("Hash for INI\\Map Code\\" + gameMode.Name + ".ini :" +
+                    Logger.Log("Hash for INI\\Map Code\\" + gameMode.Name + ".ini: " +
                         Utilities.CalculateSHA1ForFile(ProgramConstants.GamePath + "INI\\Map Code\\" + gameMode.Name + ".ini"));
+
+                    foreach (Map map in gameMode.Maps)
+                    {
+                        if (!string.IsNullOrEmpty(map.ExtraININame))
+                            extraINIFilenames.Add("INI\\Map Code\\" + map.ExtraININame);
+                    }
```

Comment:
I'm slightly worried about the performance hit this could have if a map exists in multiple game modes and the same extra INI filename gets added to the list multiple times. It's probably a non-issue in all current mods, but if some YR mods with tons of maps started making heavy use of the extra INIs, it could cause delays.

Reply:
I now changed this to a similar blanket file check as already existed for INI\Game Options so the runtime won't scale with game mode / map count. 

I am not sure why INI\Map Code check is omitted on YR release, so I didn't change that and left the GlobalCode.ini special case for YR only as a consequence.

I also took the liberty of adjusting how both the map code / game option INI hash log messages show filenames, previously it was the full filename, now it is relative to the game root like everything else.